ext/tidy/tidy.c

```diff
@@ -1024,9 +1024,8 @@ PHP_FUNCTION(tidy_parse_string)
 	tidy_instantiate(tidy_ce_doc, return_value);
 	obj = Z_TIDY_P(return_value);
 
-	TIDY_APPLY_CONFIG(obj->ptdoc->doc, options_str, options_ht);
-
-	if (php_tidy_parse_string(obj, ZSTR_VAL(input), (uint32_t)ZSTR_LEN(input), enc) == FAILURE) {
+	if (php_tidy_apply_config(obj->ptdoc->doc, options_str, options_ht) != SUCCESS
+	 || php_tidy_parse_string(obj, ZSTR_VAL(input), (uint32_t)ZSTR_LEN(input), enc) == FAILURE) {
```

Comment:
Why is one `!= SUCCESS` and the other one is `== FAILURE`? I think it would be better if both use the same comparison check.